engine/execution/ingestion/uploader/uploader_test.go

```diff
@@ -190,12 +200,15 @@ func Test_AsyncUploader(t *testing.T) {
 		async.SetOnCompleteCallback(func(computationResult *execution.ComputationResult, err error) {
 			onCompleteCallbackCalled = true
 		})
+		ctx, cancel := irrecoverable.NewMockSignalerContextWithCancel(t, context.Background())
+		async.Start(ctx)
 
 		err := async.Upload(computationResult)
 		require.NoError(t, err)
 
 		wgUploadCalleded.Wait()
```

Comment:
```suggestion
		unittest.AssertReturnsBefore(t, wgUploadCalleded.Wait, time.Second)
```
Similar to `AssertClosesBefore`, we have a utility method `AssertReturnsBefore` which will terminate the test if a function call takes too long. It's useful for `Wait`ing on a waitgroup for the same reason `AssertClosesBefore` is useful for channel operations that might hang.